clang/lib/CodeGen/CGHLSLRuntime.cpp

```diff
@@ -407,6 +429,13 @@ llvm::Value *CGHLSLRuntime::emitInputSemantic(IRBuilder<> &B,
     llvm::Function *GroupIDIntrinsic = CGM.getIntrinsic(getGroupIdIntrinsic());
     return buildVectorInput(B, GroupIDIntrinsic, Ty);
   }
+  if (D.hasAttr<HLSLSV_PositionAttr>()) {
+    if (getArch() == llvm::Triple::spirv)
+      return createSPIRVBuiltinLoad(B, CGM.getModule(), Ty, "sv_position",
+                                    /* BuiltIn::Position */ 0);
+    llvm_unreachable(
+        "Shader signature for semantics not implemented for DXIL.");
```

Comment:
this is techincally an unreachable for all targets the way you did the conditional but this should be fine because there are only two targets right now for HLSL.

Reply:
Right, rephrased the error to be more target-agnostic.